src/pages/admin/profile.tsx

```diff
@@ -0,0 +1,158 @@
+import { VerticalForm, VerticalFormButton, VerticalFormTitle, Cropper } from '@/components/common';
+import { showToast, config } from '@/lib';
+import withAccessType, { GetServerSidePropsWithAuth } from '@/lib/hoc/withAccessType';
+import { PermissionService } from '@/lib/services';
+import { PrivateProfile } from '@/lib/types/apiResponses';
+import { reportError } from '@/lib/utils';
+import { KlefkiAPI } from '@/lib/api';
+
+import type { NextPage } from 'next';
+import Image from 'next/image';
+import { SubmitHandler, useForm } from 'react-hook-form';
+
+import { ChangeEvent, useEffect, useState, useRef } from 'react';
+
+interface UploadBoardPhotoProps {
+  user: PrivateProfile;
+}
+interface FormValues {
+  photo: File;
+}
+const UpdateProfilePage: NextPage<UploadBoardPhotoProps> = ({
+  user: { firstName, lastName },
+}: UploadBoardPhotoProps) => {
+  const { handleSubmit } = useForm<FormValues>();
+  const [file, setFile] = useState<Blob | null>(null);
+  const [croppedImg, setCroppedImg] = useState<Blob | null>(null);
+  const [uploadedURL, setUploadedURL] = useState<string | null>(null);
+  const fileInputRef = useRef<HTMLInputElement>(null); // Reference to the hidden file input
+  const [previewURL, setPreviewURL] = useState<string>('');
+
+  const handleUpload = (e: ChangeEvent<HTMLInputElement>) => {
+    const uploadedFile = e.target.files?.[0] || null;
```

Comment:
You should clear `files` so that if you accidentally close the cropper, you can re-select the same image and it'll fire the change event again (currently it doesn't seem to do this). You can clear it by setting `values` to an empty string or null:

https://github.com/acmucsd/membership-portal-ui-v2/blob/4dc9393c66ef77304a91c1352455d1058d55ff10/src/components/admin/event/EventDetailsForm/index.tsx#L316-L317